### PR TITLE
remove tests/deps/

### PR DESCRIPTION
**Pull request by timotheecour**
tests/deps/ was introduced in d220809:
>  2018-09-14 00:45 rumpf_a         I disable Nimble dependencies for appveyor

but IMO this doesn't make sense:
* causes a lot of churn (lots of PR's end up (incorrectly IMO) modifying fiels that are copied from external repos; this causes one to have to write exceptions (don't modify this file because it's under tests/deps etc)

external files should not be checked in a repo, but instead any number of alternative should be used:
* submodules
* cloning in CI
* important_packages

etc

IMO important_packages subsumes `tests/deps`

## future work
- [ ] improve documentation for `--nimblePath` and error messages when misused, refs https://github.com/nim-lang/Nim/pull/17132#issuecomment-783994833
- [ ] `NIMBLE_DIR` should be set in `runCI` so that `./koch runCI` doesn't interfere with user's nimble packages

### CODE REVIEW

**Comment by Araq** [comment]
Fine with me, iirc this predates important_packages.

**Comment by timotheecour** [comment]
PTAL, finally ready.
a tricky part was that `--nimblePath` is under-documented, and behaves really obscurely (cryptic errors) when you pass in the wrong dir, eg:
* --nimblePath:build/deps/pkgs works
* --nimblePath:build/deps didn't with cryptic errors in CI

this can be improved in future work

File: tools/kochdocs.nim [hunk 1]
```diff
@@ -71,6 +71,7 @@ template inFold*(desc, body) =
 proc execFold*(desc, cmd: string, errorcode: int = QuitFailure, additionalPath = "") =
   ## Execute shell command. Add log folding for various CI services.
   # https://github.com/travis-ci/travis-ci/issues/2285#issuecomment-42724719
+  let desc = if desc.len == 0: cmd else: desc
```
Comment (timotheecour) [inline tools/kochdocs.nim:74]: unrelated change that allows passing "" for convenience (DRY)
